get_gfwData.R

```diff
@@ -1,55 +1,86 @@
-get_gfwData <- function(region, start_date, end_date, temp_res, spat_res, key, n_crs = 4326) {
-  
+get_gfwData <- function(region, start_date, end_date, temp_res,
+                        spat_res = "low",
+                        key = gfwr::gfw_auth(),
+                        cCRS = "EPSG:4326",
+                        compress = FALSE) {
+
+
   region_id <- get_region_id(region_name = region, region_source = 'eez', key = key)$id[1]
-  
+
   # Convert dates into Date objects
   start_date <- as.Date(start_date, format = "%Y-%m-%d")
   end_date <- as.Date(end_date, format = "%Y-%m-%d")
-  
+
   # Function to obtain data for a specific date range
   get_data_for_range <- function(start_date, end_date) {
     date_range <- paste(start_date, end_date, sep = ",")
-    
+
     data <- get_raster(
       spatial_resolution = spat_res,
       temporal_resolution = temp_res,
       group_by = 'flagAndGearType',
       date_range = date_range,
-      region = region_id, 
+      region = region_id,
       region_source = 'eez',
       key = key
     )
-    
+
     return(data)
   }
-  
+
   # Check whether the date range is less than or equal to 366 days
   if (as.numeric(difftime(end_date, start_date, units = "days")) <= 366) {
     # If yes, obtain data for the entire date range
-    data_sf <- get_data_for_range(start_date, end_date) %>%
-      sf::st_as_sf(coords = c("Lon", "Lat"), crs = n_crs)
+    data_df <- get_data_for_range(start_date, end_date)
+
   } else {
     # If not, divide the date range into 366-day chunks and obtain the data for each chunk.
     date_chunks <- seq(start_date, end_date, by = "366 days")
-    data_sf <- purrr::map(date_chunks, ~ get_data_for_range(.x, min(.x + 365, end_date))) %>%
-      dplyr::bind_rows() %>%
-      sf::st_as_sf(coords = c("Lon", "Lat"), crs = n_crs)
+    data_df <- purrr::map(date_chunks, ~ get_data_for_range(.x, min(.x + 365, end_date)))
   }
-  
-  # Separate the "Time Range" column based on the specified temp_res
-  if (temp_res == "yearly") {
-    data_sf <- data_sf %>%
-      dplyr::mutate(Year = `Time Range`)
-  } else {
-    # Sinon, séparer la colonne "Time Range" selon le temp_res spécifié
-    if (temp_res == "monthly") {
-      data_sf <- data_sf %>%
-        tidyr::separate("Time Range", into = c("Year", "Month"), sep = "-", remove = FALSE)
-    } else if (temp_res == "daily") {
-      data_sf <- data_sf %>%
-        tidyr::separate("Time Range", into = c("Year", "Month", "Day"), sep = "-", remove = FALSE)
+
+  if (isTRUE(compress)){
+    # GFW data will always be "EPSG:4326". No need to have CRS as an option here
```

Comment:
It seems that the PUs EPSG is 54009, so that is why I thought about transforming the crs to "fit" with the PUs of spatialplanr, but I guess for people that only want to extract some Fishing effort it's alright?

Reply:
What you were doing with `st_as_sf(....crs = )` was just telling sf what `crs` the data was in. You weren't changing it. By telling sf the data was in `ESRI:54009` when it was actually was `EPSG:4326` was just confusing sf and results in a lot of errors. So at this point you need to tell sf what crs the data is currently in. In the case of gfw, it will always be latlon.
If you want to change the crs you need to use `st_transform` which is what I do at the bottom.
Does this make sense?

Reply:
All right I understand now, thank you